cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,25 +180,30 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
+      connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, savedToken.getUsername());
+      clientConfig.setConnectionConfig(connectionInfo);
+      checkConnection(clientConfig, savedToken.getAccessToken());
+      return savedToken.getAccessToken();
     } catch (UnauthorizedException ignored) {
       // access token invalid - fall through to try acquiring token manually
     }
 
-    return getNewAccessToken(connectionInfo, output, debug);
+    return getNewAccessToken(clientConfig, output, debug);
   }
 
-  private AccessToken getNewAccessToken(ConnectionConfig connectionInfo, PrintStream output,
-                                        boolean debug) throws IOException {
+  private AccessToken getNewAccessToken(ClientConfig clientConfig,
```

Comment:
Instead of modifying `clientConfig` for setting `UserAccessToken`, you should just return `UserAccessToken`. This method isn't intended to modify anything, just return an `AccessToken`.